[wolfssl] Build wolfssl with HAVE_EX_DATA and NO_WOLFSSL_STUB to enable async use and coexistence with openssl without link error

ycdev1 opened the pull request:
Async applications using wolfssl need to associate extra data (application context) with the ssl session and context. This is used in verification callback.
NO_WOLFSSL_STUB is needed to prevent a link time error resulting from duplicate symbols when linking to both wolfssl and openssl at the same time

ycdev1 committed:
Edit portfile.cmake and baseline.josn

ycdev1 committed:
Edit versions\w-\wolfssl.json and baseline.josn

MonicaLiu0311 commented:
Please get [failure logs here](https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/103408/artifacts?artifactName=failure%20logs%20for%20x64-osx&api-version=7.1&%24format=zip) on `x64-osx`.
```
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - broken
CMake Error at /Users/vcpkg/Data/downloads/tools/cmake-3.29.2-osx/cmake-3.29.2-macos-universal/CMake.app/Contents/share/cmake-3.29/Modules/CMakeTestCCompiler.cmake:67 (message):
  The C compiler

    "/Library/Developer/CommandLineTools/usr/bin/cc"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: '/Users/vcpkg/Data/b/wolfssl/x64-osx-rel/CMakeFiles/CMakeScratch/TryCompile-PwvBMm'
    
    Run Build Command(s): /usr/local/bin/ninja -v cmTC_dfc59
```

MonicaLiu0311 commented:
Specific steps are as follows, taking `arm64-android` as an example:

1. Click "details" to enter
![image](https://github.com/microsoft/vcpkg/assets/110024546/c2185db0-e4ab-4cc7-8099-3074f9c90419)

2. Click the link for "errors/warnings"
![image](https://github.com/microsoft/vcpkg/assets/110024546/713a443b-8620-454b-809f-d02f3167c027)

3. Click the "triplets" you want to view
![image](https://github.com/microsoft/vcpkg/assets/110024546/a0a87420-2e8f-4f61-84f3-b1b6685b7389)

4. Download the log file
![image](https://github.com/microsoft/vcpkg/assets/110024546/a8152c28-12d7-429a-a2ea-4b3fa6f96627)
![image](https://github.com/microsoft/vcpkg/assets/110024546/f2cb247b-2a37-4e8c-ac46-5a45419f0e26)

5. View the stdout-triplets.log log to continue
![image](https://github.com/microsoft/vcpkg/assets/110024546/0544f036-e1a0-4833-a134-edbf3c0d9660)

dg0yt commented on ports/wolfssl/portfile.cmake:
```diff
@@ -17,7 +17,7 @@ vcpkg_cmake_get_vars(cmake_vars_file)
 
 foreach(config RELEASE DEBUG)
-  string(APPEND VCPKG_COMBINED_C_FLAGS_${config} " -DWOLFSSL_ALT_CERT_CHAINS -DWOLFSSL_DES_ECB -DWOLFSSL_CUSTOM_OID -DHAVE_OID_ENCODING -DWOLFSSL_CERT_GEN -DWOLFSSL_ASN_TEMPLATE -DWOLFSSL_KEY_GEN -DHAVE_PKCS7 -DHAVE_AES_KEYWRAP -DWOLFSSL_AES_DIRECT -DHAVE_X963_KDF")
+  string(APPEND VCPKG_COMBINED_C_FLAGS_${config} "-DHAVE_EX_DATA -DNO_WOLFSSL_STUB -DWOLFSSL_ALT_CERT_CHAINS -DWOLFSSL_DES_ECB -DWOLFSSL_CUSTOM_OID -DHAVE_OID_ENCODING -DWOLFSSL_CERT_GEN -DWOLFSSL_ASN_TEMPLATE -DWOLFSSL_KEY_GEN -DHAVE_PKCS7 -DHAVE_AES_KEYWRAP -DWOLFSSL_AES_DIRECT -DHAVE_X963_KDF")
```
```suggestion
  string(APPEND VCPKG_COMBINED_C_FLAGS_${config} " -DHAVE_EX_DATA -DNO_WOLFSSL_STUB -DWOLFSSL_ALT_CERT_CHAINS -DWOLFSSL_DES_ECB -DWOLFSSL_CUSTOM_OID -DHAVE_OID_ENCODING -DWOLFSSL_CERT_GEN -DWOLFSSL_ASN_TEMPLATE -DWOLFSSL_KEY_GEN -DHAVE_PKCS7 -DHAVE_AES_KEYWRAP -DWOLFSSL_AES_DIRECT -DHAVE_X963_KDF")
```

ycdev1 committed:
Update ports/wolfssl/portfile.cmake

missing space before -DHAVE_EX_DATA to avoid having something like '-arch x86_64-DHAVE_EX_DATA' in the command line!

Co-authored-by: Kai Pastor <dg0yt@darc.de>

ycdev1 commented:
thanks for the review! I didn't think without this space it will end up with '-arch x86_64-DHAVE_EX_DATA' on the command line as it worked on windows and linux

MonicaLiu0311 commented:
At present, the version information of your commit is inconsistent with your local one:
```
##[error]Found the following errors:
##[error]versions/w-/wolfssl.json(2,): error : while parsing versions for wolfssl from version/w-/wolfssl.json
5.7.0#1 is declared with 4c06acb4358f2d96b464a6ea1466e4b27d42bbda, but the local port has a different SHA 60802b6ec5e1284242e1b501a1eed34033bbdc35.
```
Please run `./vcpkg x-add-version testPort [--overwrite-version]` before each push:
1. For any file modification under the `ports/testPort/` folder, please commit after modification
2. Run `./vcpkg x-add-version testPort` to generate a series of version according to your commit info (Add `--overwrite-version` when generating other than the first time)
3. Commit verison information again
4. Push together

`--overwrite-version`
After the first modification, please run `./vcpkg x-add-version testPort` to generate the corresponding json; for subsequent modifications of the same version, please use `./vcpkg x-add-version testPort --overwrite-version` to update git-tree.

[maintainer guide#versioning](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md#versioning).

Commit:
update git-tree
